test/controllers/AssociateDucrSummaryControllerSpec.scala

```diff
@@ -16,86 +16,182 @@
 
 package controllers
 
-import base.{MovementBaseSpec, ViewValidator}
+import java.util.UUID
+
+import base.MockFactory._
+import base.{MockAuthConnector, URIHelper}
 import controllers.exception.IncompleteApplication
 import controllers.storage.FlashKeys
 import forms.Choice.AllowedChoiceValues
 import forms.{AssociateDucr, Choice, MucrOptions}
-import org.mockito.ArgumentMatchers._
-import org.mockito.BDDMockito._
+import org.mockito.ArgumentMatchers.{any, anyString, eq => meq}
 import org.mockito.Mockito._
-import org.scalatest.BeforeAndAfterEach
-import play.api.libs.json.Json
+import org.mockito.{InOrder, Mockito}
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{MustMatchers, WordSpec}
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.Application
+import play.api.i18n.{Messages, MessagesApi}
+import play.api.inject.bind
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.Result
+import play.api.test.FakeRequest
 import play.api.test.Helpers._
-import uk.gov.hmrc.http.HttpResponse
+import play.filters.csrf.{CSRFConfig, CSRFConfigProvider, CSRFFilter}
+import services.{CustomsCacheService, SubmissionService}
+import uk.gov.hmrc.auth.core.AuthConnector
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse, SessionKeys}
+import utils.FakeRequestCSRFSupport._
 
 import scala.concurrent.Future
 
-class AssociateDucrSummaryControllerSpec extends MovementBaseSpec with ViewValidator with BeforeAndAfterEach {
+class AssociateDucrSummaryControllerSpec
+    extends WordSpec with GuiceOneAppPerSuite with MockAuthConnector with ScalaFutures with MustMatchers
+    with URIHelper {
 
   private val uri = uriWithContextPath("/associate-ducr-summary")
 
-  override def beforeEach() {
+  private val submissionServiceMock = buildSubmissionServiceMock
+  private val customsCacheServiceMock = buildCustomsCacheServiceMock
```

Comment:
Inlining mock is better. Specially for this case. You receive some methods from this but still below you need to specify different behaviour. Comment related with stuff above.